sql/src/main/java/io/crate/user/UserActions.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.user;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.crate.analyze.CreateUserAnalyzedStatement;
+import io.crate.analyze.symbol.ParamSymbols;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.analyze.symbol.ValueSymbolVisitor;
+import io.crate.data.Row;
+import org.elasticsearch.common.settings.SecureString;
+
+import java.security.GeneralSecurityException;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiFunction;
+
+import static io.crate.concurrent.CompletableFutures.failedFuture;
+
+public final class UserActions {
+
+    private UserActions() {
+
+    }
+
+    public static CompletableFuture<Long> executeCreateUser(BiFunction<String, SecureHash, CompletableFuture<Long>> userAction,
```

Comment:
sry but still don't think the name is reflecting what this method is all about. the main thing here done is resolving the password and generating a hash out of it (and yes then applies the given BiFunction). maybe just let it generate the hash and do the execute out of this function like it was before? also this would be more correct at the exception handling as afaik the catched exception can only occur while generating the hash and not while applying the bifunction. by doing this, the related test method would also be much simpler and clear